localstack-core/localstack/services/apigateway/helpers.py

```diff
@@ -702,7 +702,8 @@ def extract_path_params(path: str, extracted_path: str) -> Dict[str, str]:
 
 def extract_query_string_params(path: str) -> Tuple[str, Dict[str, str]]:
     parsed_path = urlparse.urlparse(path)
-    path = parsed_path.path
+    if not path.startswith("//"):
+        path = parsed_path.path
```

Comment:
q: Could you explain why this is necessary / useful here?

Reply:
Sure! This is always the same issue when using `RAW_URI`, and this helper should be removed once we remove the legacy implementation, it is doing a lot of parsing it shouldn't need to do (and sometimes it does it wrong).

If you pass a path only to `urlparse`, it can properly parse it and will return the schema/netloc etc to be `None`. However, if you pass it a path that starts with multiple slashes, it will remove the first slashes.
But if it does not start with multiple slashes, we still need to use `urlparse` because sometimes, the `RAW_URI` can be a full URL (often in combination with proxies).

Here's the comments I wrote about it in `get_raw_path`:
```python
        # werkzeug/flask request (already a string, and contains the query part)
        # we need to parse it, because the RAW_URI can contain a full URL if it is specified in the HTTP request
        raw_uri: str = request.environ.get("RAW_URI", "")
        if raw_uri.startswith("//"):
            # if the RAW_URI starts with double slashes, `urlparse` will fail to decode it as path only
            # it also means that we already only have the path, so we just need to remove the query string
            return raw_uri.split("?")[0]
        return urlparse(raw_uri or request.path).path
```